### ISSUE TITLE
CVE-2021-45046

### ISSUE THREAD

**Issue by hmoffatt** [1lann/log4shelldetect#4]
Can this detect the missing fix for the second log4j issue, CVE-2021-45046?

**Comment by 1lann** [comment]
Not yet, no. It will appear as either VULNRBL (log4j < 2.15.0) or PATCHED (log4j >= 2.15.0). That issue doesn't looks quite as severe as >= 2.15.0 (PATCHED) only allows localhost JNDI servers by default so it's unlikely to be significantly exploitable unless the attacker has the ability to bind to a localhost port as well. I'll consider implementing detection when I have time.

**Comment by fboender** [comment]
Looks like the issues in 2.15 are more serious than just a DoS. From https://arstechnica.com/information-technology/2021/12/patch-fixing-critical-log4j-0-day-has-its-own-vulnerability-thats-under-exploit/:

> On Wednesday, researchers at security firm Praetorian said there’s an even more serious vulnerability in 2.15.0—an information disclosure flaw that can be used to download data from affected servers.
> [...]
> “In our research, we have demonstrated that 2.15.0 can still allow for exfiltration of sensitive data in certain circumstances,” Praetorian researcher Nathan Sportsman wrote. “We have passed technical details of the issue to the Apache Foundation, but in the interim, we strongly recommend that customers upgrade to 2.16.0 as quickly as possible.”

Not entirely sure, but from [this video](https://www.youtube.com/embed/bxDEJDqANig) it looks like they're using DNS to bypass the localhost restriction to exfiltrate data.

log4shelldetect has been instrumental in our detection of this vulnerability so far, thank you so much for this tool. It would be very helpful if it would also show 2.15 as vulnerable.

**Comment by 1lann** [comment]
@fboender I'm glad to hear that it's been useful for you.

I've added detection for Log4j 2.15 in v0.0.5 so I'll be closing this. There's a minor problem that the exit codes in `-mode list` are a bit wrong that will be fixed in `v0.0.6` but I've exhausted my API request limit to GitHub so that'll have to wait for later.

![image](https://user-images.githubusercontent.com/804919/146524472-a501aa92-220a-4a8a-9fb3-67241438d29a.png)
